presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -96,5 +105,37 @@ protected ColumnStatistics visitLiteral(Literal node, Void context)
             builder.setNullsFraction(Estimate.zeroValue());
             return builder.build();
         }
+
```

Comment:
in https://github.com/Teradata/presto/pull/573/commits/0b6441bfaf71a665132240c185154ad88b800599#diff-fb04c09ffde1723818114d064548bfd9R57 I would add `checkState` that object in lo/hi have the same type as expected for a column